Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintLocationTests.cpp

```diff
@@ -0,0 +1,339 @@
+
+        ON_CALL(mockHandler, OnPaint)
+            .WillByDefault(
+                [=](const AZ::Aabb& dirtyArea,
+                    AzToolsFramework::PaintBrushNotifications::ValueLookupFn& valueLookupFn,
+                    AzToolsFramework::PaintBrushNotifications::BlendFn& blendFn)
+                {
+                    // The OnPaint method for a listener to the PaintBrushNotificationBus should work as follows:
+                    // - It should receive a dirtyArea AABB that contains the region that's been painted.
+                    // - For each point that the listener cares about in that region, it should call valueLookupFn() to find
+                    //   out which points actually fall within the paintbrush, and what the opacities of those points are.
+                    // - For each valid point and opacity, the listener should call blendFn() to blend the values together.
+
+                    // Validate the dirtyArea AABB:
+                    // We expect the AABB to be centered around the TestBrushCenter but with a Z value of 0
+                    // because we only support painting in 2D for now. The radius of the AABB should match the radius of our paintbrush.
+                    EXPECT_THAT(dirtyArea, IsClose(AZ::Aabb::CreateCenterRadius(TestBrushCenter2d, TestBrushRadius)));
+
+                    // Validate the valueLookupFn:
+                    // Create a 3x3 square grid of points. Because our brush is a circle, we expect only the points in along a + to be
+                    // returned as valid points. The corners of the square should fall outside the circle and not get returned.
+                    const AZStd::vector<AZ::Vector3> points = {
+                        AZ::Vector3(dirtyArea.GetMin().GetX(), dirtyArea.GetMin().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetCenter().GetX(), dirtyArea.GetMin().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMax().GetX(), dirtyArea.GetMin().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMin().GetX(), dirtyArea.GetCenter().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetCenter().GetX(), dirtyArea.GetCenter().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMax().GetX(), dirtyArea.GetCenter().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMin().GetX(), dirtyArea.GetMax().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetCenter().GetX(), dirtyArea.GetMax().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMax().GetX(), dirtyArea.GetMax().GetY(), 0.0f),
+                    };
+                    AZStd::vector<AZ::Vector3> validPoints;
+                    AZStd::vector<float> opacities;
+                    valueLookupFn(points, validPoints, opacities);
+
+                    // We should only have the 5 points along the + in validPoints.
+                    EXPECT_EQ(validPoints.size(), 5);
```

Comment:
uber nit: This check is technically redundant as `:testing::Pointwise(` will fail if the number of points doesn't match but this can stay just for readability if you'd prefer it

Reply:
Oooh, neat, didn't know that! I'll remove all of them except the one that's checking for 3 valid points from 4 input points, since that's worth documenting a little more explicitly.